### PR TITLE
Scheduler PostFilter API

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind api-change
/kind feature
/sig scheduling

**What this PR does / why we need it**:

This PR introduced type `PostFilter` to scheduler component config API on version v1beta1.

**Which issue(s) this PR fixes**:

Part of #91038.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`PostFilter` type is added to scheduler component config API on version v1beta1.
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
/assign @lavalamp 

cc/ @damemi @ahg-g @alculquicondor

**Comment by alculquicondor** [comment]
/lgtm

from sig-scheduling

File: pkg/scheduler/apis/config/types.go [hunk 1]
```diff
@@ -190,6 +190,9 @@ type Plugins struct {
 	// Filter is a list of plugins that should be invoked when filtering out nodes that cannot run the Pod.
 	Filter *PluginSet
 
+	// PostFilter is a list of plugins that are invoked after filtering phase.
```
Comment (ahg-g) [inline pkg/scheduler/apis/config/types.go:193]: pls indicate that they will be invoked on both success and failure of the filtering phase.

Comment (Huang-Wei) [reply pkg/scheduler/apis/config/types.go:193]: done.

**Review by damemi — APPROVED** [review]
/lgtm

File: staging/src/k8s.io/kube-scheduler/config/v1beta1/types.go [hunk 2]
```diff
@@ -173,6 +173,9 @@ type Plugins struct {
 	// Filter is a list of plugins that should be invoked when filtering out nodes that cannot run the Pod.
 	Filter *PluginSet `json:"filter,omitempty"`
 
+	// PostFilter is a list of plugins that are invoked after filtering phase, no matter filtering succeeds or fails.
```
Comment (lavalamp) [inline staging/src/k8s.io/kube-scheduler/config/v1beta1/types.go:176]: grammar nit: "no matter whether" or "whether filtering succeeds or not"

**Comment by lavalamp** [comment]
/approve

In case you want to fix the nit:
/hold

File: staging/src/k8s.io/kube-scheduler/config/v1beta1/types.go (hunk 2 above)
Comment (Huang-Wei) [reply staging/src/k8s.io/kube-scheduler/config/v1beta1/types.go]: sure, fixed.

**Pull request merged**